src/main/java/net/glowstone/command/GameModeUtils.java

```diff
@@ -1,78 +1,136 @@
+                GameMode mode = nameToModeMap.get(name);
+                modeToNameBuilder.put(mode, name);
+                modeAutocompleteListBuilder.add(name.toLowerCase(Locale.getDefault()));
+            }
+            modeToNameMap = modeToNameBuilder.build();
+            modeAutoCompleteList = modeAutocompleteListBuilder.build();
         }
-        MODE_TO_NAME = modeToNameBuilder.build();
-        MODE_AUTOCOMPLETE_LIST = modeAutocompleteListBuilder.build();
     }
 
+    private static final LoadingCache<Locale, GameModeMaps> MAPS_CACHE = CacheBuilder.newBuilder()
+            .maximumSize(GlowVanillaCommand.CACHE_SIZE)
+            .build(CacheLoader.from(GameModeMaps::new));
+
     private GameModeUtils() {
     }
 
+    private static Locale localeFromNullable(@Nullable Locale in) {
+        return in == null ? Locale.getDefault() : in;
+    }
+
     /**
      * Create a GameMode from a string.
      *
      * @param mode The mode to convert
+     * @param locale The input locale
      * @return The matching mode if any, null otherwise.
      */
-    public static GameMode build(final String mode) {
-        if (mode == null) {
-            return null;
+    public static GameMode build(@Nullable final String mode, @Nullable final Locale locale) {
+        try {
+            return MAPS_CACHE.get(localeFromNullable(locale)).nameToMode(mode);
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
         }
-        return NAME_TO_MODE.getOrDefault(mode.toLowerCase(), null);
     }
 
     /**
      * Pretty print the given GameMode.
      *
      * @param gameMode The mode to print
+     * @param locale The output locale
      * @return A string containing the pretty name of the mode, 'Unknown' if the mode is not known,
      *     or null if the given mode is null.
      */
-    public static String prettyPrint(final GameMode gameMode) {
+    @Nullable
+    public static String prettyPrint(@Nullable GameMode gameMode, @Nullable Locale locale) {
         if (gameMode == null) {
             return null;
         }
-        return MODE_TO_NAME.getOrDefault(gameMode, GlowstoneMessages.GameMode.UNKNOWN.get());
+        try {
+            return MAPS_CACHE.get(localeFromNullable(locale)).modeToName(gameMode);
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
+        }
     }
 
+    /**
+     * Returns autocomplete suggestions that are game-mode names.
+     *
+     * @param arg The partial input
+     * @param locale The input locale
+     * @return A list of autocomplete suggestions
+     */
     @NotNull
-    public static List<String> partialMatchingGameModes(String arg) {
-        return StringUtil.copyPartialMatches(arg, MODE_AUTOCOMPLETE_LIST,
-            new ArrayList<>(MODE_AUTOCOMPLETE_LIST.size()));
+    public static List<String> partialMatchingGameModes(String arg, @Nullable Locale locale) {
```

Comment:
Same here about falling back to English.

Reply:
Done; the autocomplete list now includes the English names as well as the local ones.

Reply:
Done by adding the English names to the list when it's built.